operator/controllers/operation_controller_test.go

```diff
@@ -0,0 +1,299 @@
+				t.Errorf("reconcilePhase() = %v, want %v", tt.args.operation, tt.want.operation)
+			}
+		})
+	}
+}
+
+func TestOperatorReconciler_reconcilePause(t *testing.T) {
+	type args struct {
+		operation *operatorv1.Operation
+	}
+	type want struct {
+		operation *operatorv1.Operation
+		events    int
+	}
+	tests := []struct {
+		name string
+		args args
+		want want
+	}{
+		{
+			name: "Reconcile an Operation not paused with Spec paused",
+			args: args{
+				operation: &operatorv1.Operation{
+					Spec: operatorv1.OperationSpec{
+						Paused: true,
+					},
+					Status: operatorv1.OperationStatus{
+						Paused: false,
+					},
+				},
+			},
+			want: want{
+				operation: &operatorv1.Operation{
+					Spec: operatorv1.OperationSpec{
+						Paused: true,
+					},
+					Status: operatorv1.OperationStatus{
+						Paused: true,
+					},
+				},
+				events: 1,
+			},
+		},
+		{
+			name: "Reconcile an Operation paused with Spec not paused",
+			args: args{
+				operation: &operatorv1.Operation{
+					Spec: operatorv1.OperationSpec{
+						Paused: false,
+					},
+					Status: operatorv1.OperationStatus{
+						Paused: true,
+					},
+				},
+			},
+			want: want{
+				operation: &operatorv1.Operation{
+					Spec: operatorv1.OperationSpec{
+						Paused: false,
+					},
+					Status: operatorv1.OperationStatus{
+						Paused: false,
+					},
+				},
+				events: 1,
+			},
+		},
+		{
+			name: "Reconcile an Operation paused with Spec paused",
+			args: args{
+				operation: &operatorv1.Operation{
+					Spec: operatorv1.OperationSpec{
+						Paused: true,
+					},
+					Status: operatorv1.OperationStatus{
+						Paused: true,
+					},
+				},
+			},
+			want: want{
+				operation: &operatorv1.Operation{
+					Spec: operatorv1.OperationSpec{
+						Paused: true,
+					},
+					Status: operatorv1.OperationStatus{
+						Paused: true,
+					},
+				},
+				events: 0,
+			},
+		},
+		{
+			name: "Reconcile an Operation not paused with Spec not paused",
+			args: args{
+				operation: &operatorv1.Operation{
+					Spec: operatorv1.OperationSpec{
+						Paused: false,
+					},
+					Status: operatorv1.OperationStatus{
+						Paused: false,
+					},
+				},
+			},
+			want: want{
+				operation: &operatorv1.Operation{
+					Spec: operatorv1.OperationSpec{
+						Paused: false,
+					},
+					Status: operatorv1.OperationStatus{
+						Paused: false,
+					},
+				},
+				events: 0,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			rec := record.NewFakeRecorder(1)
+
+			r := &OperationReconciler{
+				recorder: rec,
+			}
+
+			r.reconcilePause(tt.args.operation)
+
+			if !reflect.DeepEqual(tt.args.operation, tt.want.operation) {
+				t.Errorf("reconcilePause() = %v, want %v", tt.args.operation, tt.want.operation)
+			}
+
+			if tt.want.events != len(rec.Events) {
+				t.Errorf("reconcilePause() = %v events recorded, want %v events", tt.want.events, len(rec.Events))
```

Comment:
no need to have the name of the function here. the Test* name already points where this is coming from.